features/include/pcl/features/impl/our_cvfh.hpp

```diff
@@ -495,14 +495,11 @@ pcl::OURCVFHEstimation<PointInT, PointNT, PointOutT>::computeRFAndShapeDistribut
             weights[ii] *= 0.5f - wz * 0.5f;
         }
 
-        int h_index = (d <= 0) ? 0 : std::ceil (size_hists * (d / distance_normalization_factor)) - 1;
-        /* from http://www.pcl-users.org/OUR-CVFH-problem-td4028436.html
-           h_index will be 13 when d is computed on the farthest away point.
-
-          adding the following after computing h_index fixes the problem:
-        */
-        if(h_index > 12)
-          h_index = 12;
+        const int raw_index = static_cast<int>(std::floor (size_hists * (d / distance_normalization_factor)));
```

Comment:
Assuming you want the following mapping:
d / distance_normalization_factor  = 0; index = 0
d / distance_normalization_factor = 1; index = size_hists - 1 

my recommendation is use `std::round` and the following linear transformation
```cpp
std::round ((size_hists - 1) * (d / distance_normalization_factor))
```

followed by the cast to int.

Reply:
Round sounds better. I'd modify the similar looking assignments in VFH also (same logic is used there). It'd also reduce the next clamp (lower 0, upper: `size-1`)

Reply:
Wouldn't this cause the first and last buckets to be half as full? All the intermediate buckets will get points from both the sides, but the first and last buckets can get them from one side only

Reply:
Yes it will.

Reply:
Will this have any impact on result/performance? I've not read the theory behind this algorithm. Is the increased non-uniformity in bucket density offset by better quantization error?